### PR TITLE
Create read and write access level roles and give grants

### PR DESCRIPTION
**Pull request by blagojts**
The prom_reader and prom_writer roles are created
and granted specific access to all the different schemas,
tables and functions. prom_writer inherits from prom_reader.
prom_reader has read-only access.

### CODE REVIEW

File: pkg/pgmodel/migrations/sql/1_base_schema.up.sql [hunk 1]
```diff
@@ -701,6 +778,7 @@ $$
 LANGUAGE SQL VOLATILE;
 COMMENT ON FUNCTION SCHEMA_PROM.set_default_retention_period(INTERVAL)
 IS 'set the retention period for any metrics (existing and new) without an explicit override';
+GRANT EXECUTE ON FUNCTION SCHEMA_PROM.set_default_retention_period(INTERVAL) TO prom_writer;
```
Comment (cevian) [inline pkg/pgmodel/migrations/sql/1_base_schema.up.sql:781]: this should be an admin func

File: pkg/pgmodel/migrations/sql/1_base_schema.up.sql [hunk 2]
```diff
@@ -715,6 +793,8 @@ AS $func$
     SELECT true;
 $func$
 LANGUAGE SQL VOLATILE;
+GRANT EXECUTE ON FUNCTION SCHEMA_PROM.set_metric_retention_period(TEXT, INTERVAL) TO prom_writer;
```
Comment (cevian) [inline pkg/pgmodel/migrations/sql/1_base_schema.up.sql:796]: admin function

File: pkg/pgmodel/migrations/sql/1_base_schema.up.sql [hunk 3]
```diff
@@ -728,6 +808,7 @@ $func$
 LANGUAGE SQL VOLATILE;
 COMMENT ON FUNCTION SCHEMA_PROM.reset_metric_retention_period(TEXT)
 IS 'resets the retention period for a specific metric to using the default';
+GRANT EXECUTE ON FUNCTION SCHEMA_PROM.reset_metric_retention_period(TEXT) TO prom_writer;
```
Comment (cevian) [inline pkg/pgmodel/migrations/sql/1_base_schema.up.sql:811]: admin func

File: pkg/pgmodel/migrations/sql/1_base_schema.up.sql [hunk 4]
```diff
@@ -827,6 +907,7 @@ BEGIN
 END
 $func$
 LANGUAGE PLPGSQL VOLATILE;
+GRANT EXECUTE ON FUNCTION SCHEMA_CATALOG.drop_metric_chunks(TEXT, TIMESTAMPTZ) TO prom_writer;
```
Comment (cevian) [inline pkg/pgmodel/migrations/sql/1_base_schema.up.sql:910]: admin func

File: pkg/pgmodel/migrations/sql/1_base_schema.up.sql [hunk 5]
```diff
@@ -878,6 +960,7 @@ END;
 $$ LANGUAGE PLPGSQL;
 COMMENT ON PROCEDURE SCHEMA_PROM.drop_chunks()
 IS 'drops data according to the data retention policy. This procedure should be run regularly in a cron job';
+GRANT EXECUTE ON PROCEDURE SCHEMA_PROM.drop_chunks() TO prom_writer;
```
Comment (cevian) [inline pkg/pgmodel/migrations/sql/1_base_schema.up.sql:963]: admin func

File: pkg/pgmodel/migrations/sql/1_base_schema.up.sql [hunk 6]
```diff
@@ -1,13 +1,66 @@
...
+
 CREATE SCHEMA IF NOT EXISTS SCHEMA_INFO;
+GRANT USAGE ON SCHEMA SCHEMA_INFO TO prom_reader;
+GRANT SELECT ON ALL TABLES IN SCHEMA SCHEMA_INFO TO prom_reader;
+ALTER DEFAULT PRIVILEGES IN SCHEMA SCHEMA_INFO GRANT SELECT ON TABLES TO prom_reader;
+GRANT CREATE ON SCHEMA SCHEMA_INFO TO prom_writer;
+GRANT ALL PRIVILEGES ON ALL TABLES IN SCHEMA SCHEMA_INFO TO prom_writer;
+ALTER DEFAULT PRIVILEGES IN SCHEMA SCHEMA_INFO GRANT ALL PRIVILEGES ON TABLES TO prom_writer;
+
+GRANT prom_reader TO prom_writer;
```
Comment (davidkohn88) [inline pkg/pgmodel/migrations/sql/1_base_schema.up.sql:63]: Doesn't horribly matter, but I'd put the grant of prom reader to prom writer right after you create the roles, just makes everything else make more sense.

**Comment by blagojts** [comment]
Committed changes per comment suggestions. Additionally:

1. Also assumed set_chunk_interval functions are admin as set_retention
2. Granted only select on SCHEMA_METRIC since it has only views as SCHEMA_SERIES

**Pull request merged**